File: share/RunScript/ADGRoomNames.cs
using Autodesk.Revit.DB;

public static void Main(Document doc)
{
    using (var t = new Transaction(doc))
    {
        t.Start("Run Script");
        var fec = new FilteredElementCollector(doc).OfCategory(BuiltInCategory.OST_Rooms);
        foreach (var r in fec)
        {
            var room = r as Autodesk.Revit.DB.Architecture.Room;
            //convert sq feet to sqm
            var area = room.Area * 0.092903;
            if (area < 50) {
                room.Name = "Room";
                continue;
            }
            if (area < 75) {
                room.Name = "1B";
                continue;
            }
            if (area < 95) {
                room.Name = "2B";
                continue;
            }
            if (area >= 95) {
                room.Name = "3B";
                continue;
            }
        }
        t.Commit();
    }
}

File: share/RunScript/ADGRoomNames.lua
import ('RevitAPI', 'Autodesk.Revit.DB')

local doc = commandData.Application.ActiveUIDocument.Document
f = fec:OfCategory(BuiltInCategory.OST_Rooms):ToElements()

t     = Transaction(doc)
t:Start('ADG rooms names')
for i = 0, f.Count-1 do
        room = f[i]
        area = room.Area * 0.092903              -- convert sq feet to sqm
        if area < 50 then
              room.Name = "Room"
        elseif area < 75 then
              room.Name = "1B"
        elseif area < 95 then
              room.Name = "2B"
        elseif area >= 95 then
              room.Name = "3B"
        end
end
t:Commit()
t:Dispose()

return 'success'

   
